aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -215,6 +250,11 @@ static inline void launch_vectorized_kernel(
   if constexpr (sizeof(cpp_type) < 2) {
     vec_size = std::min<uint16_t>(vec_size, 4);
   }
+  // Since we are not compiling vec8 kernel on sm<90, we are not calling it.
+  auto dprop = at::cuda::getCurrentDeviceProperties();
+  if (dprop->major < 9) {
```

Comment:
~How much does it increase compilation times? This is an very hot codepath and is used for primitives including copying ~ IS the original PR description wrong? This won't affect compile times since it's not constexpr.

After examining where this call is used, if I am not mistaken; it's only used for vectorized copy of fp8 and lower dtypes. That is the reason why  Why would this decrease/increase compilation time? Wouldn't those float8 kernels just not be built on lower SMs? Except SM89?

Do we actually not want to allow this on FP8 supporting consumer GPUs like SM89?!
```suggestion
  if (dprop->major < 9 || (dprop->major == 8 && dprop->minor >= 9)) {
```

Reply:
> ~How much does it increase compilation times?

I would like ask @atalman about this.

>it's only used for vectorized copy of fp8 and lower dtypes

Yes, it is for float16, bfloat16 and fp8.

Reply:
In general, TensorIterator is used for elementwise ops like (`relu`, `exp`, `log`, `sigmoid`, `sin`, `cos` etc.) on all `dtypes`. Though, not all ops go trough TensorIterator.

Reply:
@Aidyn-A triggered the windows builds, lets see if there is an improvement

Reply:
Hi @Aidyn-A I don't see improvement over the nightly build time (Build Pytorch Binary):

This PR:
https://github.com/pytorch/pytorch/actions/runs/13677007625/job/38239672663?pr=148320

Nightly 
https://github.com/pytorch/pytorch/actions/runs/13759231430/job/38471718241

Roughtly 
cuda 11.8 - cuda 12.6 : 3h:40m-3h:55m and cuda 12.8 4h:20m

This is run of https://github.com/pytorch/pytorch/pull/147455
https://github.com/pytorch/pytorch/actions/runs/13413313017/job/37468143832?pr=147455
cuda 11.8 - cuda 12.6 : 3h:20m-3h:45m